contracts/v2/sovereignChains/AggOracleManager.sol

```diff
@@ -0,0 +1,307 @@
+                    lastVotedReport.votes--;
+                }
+            }
+        }
+
+        Report memory currentVotedReport = proposedGERToReport[
+            proposedGlobalExitRoot
+        ];
+
+        // Check if it's a new report
+        if (currentVotedReport.timestamp == 0) {
+            // It's a new report, set slot and votes
+            currentVotedReport.timestamp = uint64(block.timestamp);
+            currentVotedReport.votes = 1;
+        } else {
+            // It's an existing report, add a new vote
+            currentVotedReport.votes++;
+        }
+
+        // Emit Submit report before check the quorum
+        emit ProposedGlobalExitRoot(proposedGlobalExitRoot, msg.sender);
+
+        // Check if it reaches the quorum
+        if (currentVotedReport.votes >= quorum) {
+            delete proposedGERToReport[proposedGlobalExitRoot];
+
+            // Consolidate report
+            globalExitRootManagerL2Sovereign.insertGlobalExitRoot(
+                proposedGlobalExitRoot
+            );
+            emit ConsolidatedGlobalExitRoot(proposedGlobalExitRoot);
+        } else {
+            // Store submitted report with a new added vote
+            proposedGERToReport[proposedGlobalExitRoot] = currentVotedReport;
+
+            // Store voted report hash
+            addressToLastProposedGER[msg.sender] = proposedGlobalExitRoot;
+        }
+    }
+
+    ////////////////////////
+    // Owner functions
+    ////////////////////////
+
+    /**
+     * @notice Add an oracle member.
+     * Only the owner can call this function.
+     * @param newOracleMember Address of the new oracle member
+     */
+    function addOracleMember(address newOracleMember) external onlyOwner {
+        _addOracleMember(newOracleMember);
+    }
+
+    /**
+     * @notice Internal function to add an oracle member.
+     * @param newOracleMember Address of the new oracle member
+     */
+    function _addOracleMember(address newOracleMember) internal {
+        require(
+            newOracleMember != address(0),
+            OracleMemberCannotBeZero()
+        );
+
+        require(
+            addressToLastProposedGER[newOracleMember] == bytes32(0),
+            AlreadyOracleMember()
+        );
+
+        // Add oracle member
+        addressToLastProposedGER[newOracleMember] = INITIAL_PROPOSED_GER;
+
+        // Add oracle member to the oracleMembers array
+        aggOracleMembers.push(newOracleMember);
+
+        emit AddAggOracleMember(newOracleMember);
+    }
+
+    /**
+     * @notice Remove an oracle member.
+     * Only the owner can call this function.
+     * @param oracleMemberAddress Address of the oracle member to remove
+     * @param oracleMemberIndex Index of the oracle member to remove
+     */
+    function removeOracleMember(
+        address oracleMemberAddress,
+        uint256 oracleMemberIndex
+    ) external onlyOwner {
+        bytes32 lastVotedReportHash = addressToLastProposedGER[
```

Comment:
The name `lastVotedReportHash` is a bit confusing, maybe rename it to `lastProposedGER`